### ISSUE TITLE
Loading offline content displays file contents rather that loading HTML

### ISSUE THREAD

**Issue by digi-chris** [resin-io-playground/resin-wpe#6]
Hi,

I followed the instructions in the readme with regards to loading offline content from the Docker container, adding the following lines to the Dockerfile:

```
COPY public_html /var/lib/public_html

ENV WPE_URL="file:///var/lib/public_html/index.html"
```

Plus, adding a public_html/index.html file and committing it to my own fork. I pushed this to my resin master and all looks good - it boots up and loads my file. Only problem is, it displays the file contents rather than rendering the HTML.

I think this could be happening because it's just loading the file from the disk, and there are no headers being sent to the browser - no MIME content type, etc. I'm assuming that the best approach would be to install a web server of some description, but this is where I'm getting totally stumped. I'm new to Docker and Resin, I can't seem to install any type of web server into the Docker container because it doesn't have a package manager or compiler, and if it's possible to add the web server in as a submodule to the git repository, I'm just not sure how to do it!

Any help would be greatly appreciated.

Thanks,

Chris.

**Comment by matthewlymer36** [comment]
I too have this problem.

Shouldn't need a web server to solve though!

**Comment by petrosagg** [comment]
Unfortunately an HTTP server seems to be the way to go. Looking at the available http servers I think something simple like [sthttpd](https://layers.openembedded.org/layerindex/recipe/4696/) will do the job

**Comment by petrosagg** [comment]
@digi-chris I actually tried this again on the latest master image and it seems to be working. Can you try again and if it doesn't work post the html file you're using?

**Comment by digi-chris** [comment]
@petrosagg thanks, it's a while since I tested this before, but I'll give it another try when I get a chance.

**Comment by calebsauder** [comment]
@petrosagg I'm using your repo, but I've been unable to install a web-server and php which I need to run an offline script.  Could you point me in the right direction on how to accomplish this.  It seems WPE will not allow you to run apt-get within the Docker file.  Thanks!

**Comment by petrosagg** [comment]
@calebsauder the base image of this project is built with yocto, this is why apt-get is failing. This only works on ubuntu/debian base images.

To install `php` and `apache2` you'll have to add those two package names in [resin-wpe-image.bb](https://github.com/resin-io-playground/resin-wpe/blob/master/base-image/meta-resin-wpe/recipes-core/images/resin-wpe-image.bb#L5) and rebuild the base-image by [following the instructions](https://github.com/resin-io-playground/resin-wpe/blob/master/base-image/README.md). After creating the image you'll have to edit the main [Dockerfile](https://github.com/resin-io-playground/resin-wpe/blob/master/Dockerfile#L1) of this repo to use that image and also change [wpe-init](https://github.com/resin-io-playground/resin-wpe/blob/master/wpe-init) to start your webserver.

Unfortunately this is a lot more complicated than it needs to be. The proper solution is multicontainer application which are going to be released very soon in resin. That will allow resin-wpe to be a standalone container that only runs the browser and everything else will run on a normal debian based container with the two communicating over the network. I'll probably create an example here as soon as the feature is out.

**Comment by csmjulian** [comment]
@petrosagg I'm working with @calebsauder to accomplish our task. We got the build to be successful but we are not sure how to reference it properly. This is the name we received: resin-wpe:raspberrypi3-3e3af13-dirty . How do we properly reference this new base-image for our project?

**Comment by petrosagg** [comment]
@csmjulian you need to push the resulting docker image to a registry and then use it in the `FROM` of your Dockerfile. When I build an image I push it to `petrosagg/resin-wpe` and then I update the main Dockerfile like in this commit https://github.com/resin-io-playground/resin-wpe/commit/3e3af13b4db41d4f3d2a513319ed124d75b1f4d6

**Comment by csmjulian** [comment]
@petrosagg Ah I see. I assumed I could just push to a public registry but the main component I was missing was a Docker account. I was able to successfully modify the base image to load Apache, PHP, and other components our project needed. You are definitely right that it is more complicated than it needs to be but we got it to work.

As for the comment above, I was able to verify that after installing apache (specifically the apache2 module) through this new image, I was able to load local HTML files without a problem using the method you provided. I didn't test if I could before creating this new image, because I had the need to load PHP files so it didn't matter then.

When it comes to any other method of loading HTML (either through PHP or any other backend lanaguage) there seems to be an issue but it's probably due to a need to edit the configuration for Apache. If you have any tips, I'd greatly appreciate it. I had to learn all this on the fly.
